### PR TITLE
Release 0.17.0.1

### PR DESCRIPTION
**Pull request by laanwj**
Update release notes and bump version

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -16,72 +13,30 @@ To receive security and update notifications, please subscribe to:
...
-Known issues
-============
-
-...
-
 Notable changes
 ===============
 
-...
+An issue was solved with OSX dmg generation, affecting OSX 10.12 to 10.14,
```
Comment (hebasto) [inline doc/release-notes.md]: OSX --> macOS

**Comment by MarcoFalke** [comment]
ACK, if this is meant to only fix the macos build.

**Comment by Sjors** [comment]
ACK dc4923a once Travis is happy (linter failure)

**Comment by fanquake** [comment]
utACK dc4923a

Relevant IRC discussion, on wether to wait for 0.17.1 or do a quick 0.17.0.1 release is [here](https://botbot.me/freenode/bitcoin-core-dev/2018-10-25/?msg=105732007&page=5). 
Consensus seems to be that a 0.17.0.1 release is preferred, there are still quite a few outstanding PRs before 0.17.1 would be done.

The linter "failure" can be fixed by rebaseing on the 0.17 branch now that #14579 is merged.

**Pull request merged**